### DISCUSSION TITLE
Support registering additional GLTF extensions

### DISCUSSION THREAD

**Discussion by merlosy** [google/model-viewer#3353] [Ideas]
### Description

I am currently trying to build a small project to try to extend the model-viewer to support more gltf extensions on load (MSFT_audio_emitter).
I had some challenges on several aspects:
- Doing some clean up while unregistering a GLTF extension callback. It involves keeping a reference to the extension instance to be able to call some "dispose/destroy" method (in my case to call URL.revokeObjectURL). Maybe there is a hook/event that can be used to do it more cleanly?
- Make the ModelScene extendable to add extra methods and/or properties

The plan is to be fully backward compatible. Is it something that you are open to?

The other option would be to open a PR to integrate it natively (like in [Babylonjs](https://github.com/BabylonJS/Babylon.js/blob/c946425dd8e6ee0ef242b5b78ecb55985cf372f1/loaders/src/glTF/2.0/Extensions/MSFT_audio_emitter.ts)) but I think it would miss an opportunity for people to build on top this great project.

#### Live Demo

Nothing working at this stage

### Version

- model-viewer: v1.10.1

### Browser Affected

- [ ] Chrome, version: xx.x.xxxx.xx
- [ ] Edge
- [ ] Firefox
- [ ] IE
- [ ] Safari

### OS

- [ ] Android
- [ ] iOS
- [ ] Linux
- [ ] MacOS
- [ ] Windows

### AR

- [ ] WebXR
- [ ] SceneViewer
- [ ] QuickLook

**Comment by elalish** [comment]
Thank you, PRs certainly welcome! ModelScene is not part of our public API, so that might not be the best place to add things (though I may have misunderstood your intention). You may want to follow our other loader APIs like dracoDecoderLocation; note they are static, since the loader is shared among MV elements, and their style allows them to be set before the element starts pulling down the `src`, which is often pretty immediate.

**Comment by merlosy** [comment]
I'd like to start working on this if no one does in the meantime :)
The only constraint is that I need to find some time.
